pkg/module/drift/terraform.go

```diff
@@ -215,23 +205,30 @@ func (d *Drift) terraformDebugDifferentResources(ctx context.Context, conn *pgxp
 		return table
 	}
 
-	for k, tfAttrs := range tfAttList {
-		cloudAttrs, ok := cloudAttList[k]
+	tfMap, cloudMap := tfRes.Map(), differentIDs.Map()
```

Comment:
Feels like the ResourceList structure could have been defined more effectively perhaps? I noticed in queryIntoResourceList we are creating them we can add them into a smarter data structure that has functions such as Get(id)/ Lookup(type) -> Resources etc' and even a generic compare method between to Resources data structures?
We can do open this in a sperate issue/PR just food for taught while looking at the code.